bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,380 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EgateHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Frieso Aeschbacher - Initial contribution
+ */
+@NonNullByDefault
+public class eGateHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(eGateHandler.class);
+    private @Nullable Socket egateSocket;
+
+    private int port;
+    private @Nullable String host;
+    private static final int SOCKET_TIMEOUT = 250;
+    private final Object lock = new Object();
+    private @Nullable BufferedWriter writer;
+    private @Nullable BufferedReader reader;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable Map<String, ThingUID> registeredBlinds;
+    private final ScheduledExecutorService scheduler;
+
+    public eGateHandler(Bridge thing) {
+        super(thing);
+        registeredBlinds = new HashMap<String, ThingUID>();
+        scheduler = Executors.newScheduledThreadPool(1);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(GETCONFIG)) {
+            sendCommand("EthernetGet;\r");
+        }
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void initialize() {
+        logger.debug("Initializing the Dominoswiss EGate bridge handler.");
+
+        Configuration config = this.getConfig();
+        host = (String) config.get("ipAddress");
+        port = 1318; // default Port
+        if (!config.get("port").toString().isEmpty()) {
+            port = Integer.parseInt(config.get("port").toString()); // take configured Port
+        }
+
+        if (host != null && port > 0) {
+            // Create a socket to eGate
+            synchronized (this.lock) {
+                try {
+                    egateSocket = new Socket();
+                    egateSocket.connect(new InetSocketAddress(host, port));
+                    egateSocket.setSoTimeout(SOCKET_TIMEOUT);
+                    writer = new BufferedWriter(new OutputStreamWriter(egateSocket.getOutputStream()));
+                    writer.write("SilenceModeSet;Value=0;" + CR);
+                    writer.flush();
+
+                } catch (UnknownHostException e) {
+                    logger.error("unknown socket host {}", host);
+                    try {
+                        egateSocket.close();
+                        updateStatus(ThingStatus.OFFLINE);
+                    } catch (IOException e1) {
+                        if (logger.isErrorEnabled()) {
+                            logger.error("EGate Socket not closed {}", e1.toString());
+                        }
```

Comment:
you dont need to check if isErrorEnabled() for only 1 logger, it is used if you have multiple log statements in a row so then the check is only done once.

Secondly error level logging should only be done if something critical happens to the core.
See https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Fixed in new commit thanks for that hint